client/client.go

```diff
@@ -127,44 +145,33 @@ func (r *NotaryRepository) Initialize(rootKeyID string) error {
 		return fmt.Errorf("invalid format for root key: %s", privKey.Algorithm())
 	}
 
-	// All the timestamp keys are generated by the remote server.
-	remote, err := getRemoteStore(r.baseURL, r.gun, r.roundTrip)
-	if err != nil {
-		return err
-	}
-	rawTSKey, err := remote.GetKey("timestamp")
-	if err != nil {
-		return err
-	}
-
-	timestampKey, err := data.UnmarshalPublicKey(rawTSKey)
-	if err != nil {
-		return err
-	}
-
-	logrus.Debugf("got remote %s timestamp key with keyID: %s", timestampKey.Algorithm(), timestampKey.ID())
-
-	// This is currently hardcoding the targets and snapshots keys to ECDSA
-	// Targets and snapshot keys are always generated locally.
-	targetsKey, err := r.CryptoService.Create("targets", data.ECDSAKey)
-	if err != nil {
-		return err
-	}
-	snapshotKey, err := r.CryptoService.Create("snapshot", data.ECDSAKey)
+	kdb := keys.NewDB()
+	err = addKeyForRole(kdb, data.CanonicalRootRole, rootKey)
 	if err != nil {
 		return err
 	}
 
-	kdb := keys.NewDB()
-
-	kdb.AddKey(rootKey)
-	kdb.AddKey(targetsKey)
-	kdb.AddKey(snapshotKey)
-	kdb.AddKey(timestampKey)
-
-	err = initRoles(kdb, rootKey, targetsKey, snapshotKey, timestampKey)
-	if err != nil {
-		return err
+	for role, isManaged := range rolesAreManaged {
+		var key data.PublicKey
+		if isManaged {
+			// This key is generated by the remote server.
+			key, err = getRemoteKey(r.baseURL, r.gun, role, r.roundTrip)
+			if err != nil {
+				return err
+			}
+			logrus.Debugf("got remote %s %s key with keyID: %s",
+				role, key.Algorithm(), key.ID())
+		} else {
+			// This is currently hardcoding the keys to ECDSA.
+			key, err = r.CryptoService.Create(role, data.ECDSAKey)
```

Comment:
Wondering if we should explicitly create all the local keys first so that if there's an error there we haven't unnecessarily created keys on the server?